compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -2970,6 +2970,10 @@ public BLangNode transform(RequiredParameterNode requiredParameter) {
         simpleVar.pos = getPosition(requiredParameter);
         if (requiredParameter.paramName().isPresent()) {
             simpleVar.name.pos = getPosition(requiredParameter.paramName().get());
+        } else if (simpleVar.name.pos == null) {
+            // Param doesn't have a name and also is not a missing node
+            // Therefore, assigning the location of the node for the location of the var-name
+            simpleVar.name.pos = simpleVar.pos;
```

Comment:
Shall we set `symTable.builtinPos` instead since in this case, name is a compiler-generated one?

Reply:
Though it is compiler-generated, shouldn't the position of the `name` be the same as the `simpleVar`? 
When looking up for the symbol in `SymbolFinder`, we evaluate the position of the `name` (checking `withInRange`), if the position is a `builtInPos`, we get `false` for this. 🤔

Reply:
Yes, exactly. That's the idea, to ensure that `withinRange()` evaluates to `false` for compiler-generated stuff. 🙂